app/forms/NirmsNumberFormProvider.scala

```diff
@@ -27,7 +26,24 @@ class NirmsNumberFormProvider @Inject() extends Mappings {
   def apply(): Form[String] =
     Form(
       "value" -> text("nirmsNumber.error.required")
-        .transform(removeWhitespace, identity[String])
+        .transform(removeHyphensAndSpaces, identity[String])
         .verifying(regexp(StringFieldRegex.nirmsRegex, "nirmsNumber.error.invalidFormat"))
+        .transform(addHyphens, identity[String])
     )
+
+  def removeHyphensAndSpaces: String => String = _.replaceAll(" ", "")
```

Comment:
should this be still named removedWhiteSpace, as the function only remove spaces? Or it is the way round? the function should also remove hyphen (-)?

Reply:
good spot!